bugbug/models/rcatype.py

```diff
@@ -0,0 +1,167 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import logging
+import re
+
+import numpy as np
+import xgboost
+from sklearn.compose import ColumnTransformer
+from sklearn.feature_extraction import DictVectorizer
+from sklearn.multiclass import OneVsRestClassifier
+from sklearn.pipeline import Pipeline
+
+from bugbug import bug_features, bugzilla, feature_cleanup
+from bugbug.model import BugModel
+
+# For the moment, rca - XYZ is treated of bugtype XYZ,
+# so we don't need to store it in a dictionary.
+RCA_CATEGORIES = [
+    "requirementerror",
+    "poorarchitecture",
+    "designerror",
+    "codingerror",
+    "testingerror",
+    "externalsoftwareaffectingfirefox",
+    "performanceerror",
+    "standards",
+    "systemerror",
+    "localizationerror",
+    "memory",
+    "infrastructure/builderror",
+    "communicationissues",
+    "productdecision",
+    "undocumentedchange",
+    "cornercase",
+]
+
+RCA_SUBCATEGORIES = [
+    "codingerror-syntaxerror",
+    "codingerror-logicalerror",
+    "codingerror-semanticerror",
+    "codingerror-runtimeerror",
+    "codingerror-unhandledexceptions",
+    "codingerror-internalapiissue",
+    "codingerror-networkissue",
+    "codingerror-compatibilityissue",
+    "codingerror-other",
+]
+
+logger = logging.getLogger(__name__)
+
+
+class RCATypeModel(BugModel):
+    def __init__(
+        self, lemmatization=False, historical=False, rca_subcategories_enabled=False
+    ):
+        BugModel.__init__(self, lemmatization)
+
+        self.calculate_importance = False
+
+        # should we consider only the main category or all sub categories
+        self.RCA_TYPES = (
+            RCA_SUBCATEGORIES if rca_subcategories_enabled else RCA_CATEGORIES
+        )
+
+        self.RCA_LIST = sorted(set(self.RCA_TYPES))
+
+        feature_extractors = [
+            bug_features.has_str(),
+            bug_features.severity(),
+            # Ignore keywords that would make the ML completely skewed
+            # (we are going to use them as 100% rules in the evaluation phase).
+            bug_features.keywords(self.RCA_LIST),
```

Comment:
The RCA values are not keywords, they should instead be ignored from the whiteboard (if you want we can do it in a follow-up, but for now we'd need to comment out the whiteboard feature)